src/backend/utils/adt/like_match.c

```diff
@@ -350,15 +350,26 @@ do_like_escape(text *pat, text *esc)
 
 	if (elen == 0)
 	{
-		/*
-		 * No escape character is wanted.  Double any backslashes in the
-		 * pattern to make them act like ordinary characters.
-		 */
-		while (plen > 0)
-		{
-			if (*p == '\\')
-				*r++ = '\\';
-			CopyAdvChar(r, p, plen);
+		if (sql_dialect == SQL_DIALECT_TSQL){
+            /*
+            * Escape string is empty, just show the error.
```

Comment:
Also, follow the format we use for comments. It should be:

/*
 * Escape string is empty, just show the error.
 * /